File: leaflet-view/src/main/java/com/gurunars/leaflet_view/Finder.java
package com.gurunars.leaflet_view;

import java.util.List;

public class Finder {

    public interface Equator<T> {
        boolean equal(T one, T two);
    }

    public static <T> boolean contains(List<? extends T> list, T item, Equator<T> equator){
        return indexOf(list, item, equator) > 0;
    }

    public static <T> int indexOf(List<? extends T> list, T item, Equator<T> equator){
        for(int i=0; i < list.size(); i++) {
            T cursor = list.get(i);
            if (equator.equal(cursor, item)) {
                return i;
            }
        }
        return -1;
    }

}

File: leaflet-view/src/main/kotlin/com/gurunars/leaflet_view/Finder.kt
package com.gurunars.leaflet_view

internal object Finder {

    interface Equator<T> {
        fun equal(one: T, two: T): Boolean
    }

    fun <T> contains(list: List<T>, item: T, equator: Equator<T>): Boolean {
        return indexOf(list, item, equator) > 0
    }

    fun <T> indexOf(list: List<T>, item: T, equator: Equator<T>): Int {
        for (i in list.indices) {
            val cursor = list[i]
            if (equator.equal(cursor, item)) {
                return i
            }
        }
        return -1
    }

}
